File: backend/playlytics/src/main/java/com/mparker/playlytics/dto/GamePlaySessionResponseDTO.java
package com.mparker.playlytics.dto;

// Imports
import com.mparker.playlytics.enums.ScoringModel;
import java.time.Instant;
import java.util.Set;

public record GamePlaySessionResponseDTO(Long id, Instant sessionDateTime, ScoringModel scoringModel, String gameName, Set<Long> sessionParticipantIds, Set<Long> sessionTeamIds) {
}

File: frontend/src/app/dtos/game-play-session-response-dto.ts
import {ScoringModelEnum} from '../enums/scoring-model-enum';

import {PlayerResponseDTO} from './PlayerResponseDTO';

export interface GamePlaySessionResponseDTO {
  id: number;
  sessionDateTime: string;
  scoringModel: ScoringModelEnum;
  gameName: string;
  sessionParticipantDetails: Array<PlayerResponseDTO>;
}
